integration_tests/test_mempool.py

```diff
@@ -0,0 +1,87 @@
+from pathlib import Path
+
+import pytest
+from web3 import Web3
+
+from .network import setup_custom_cronos
+from .utils import (
+    ADDRS,
+    CONTRACTS,
+    KEYS,
+    deploy_contract,
+    sign_transaction,
+    wait_for_new_blocks,
+)
+
+pytestmark = pytest.mark.mempool
+
+
+@pytest.fixture(scope="module")
+def cronos(tmp_path_factory):
+    path = tmp_path_factory.mktemp("cronos-mempool")
+    cfg = Path(__file__).parent / "configs/long_timeout_commit.yaml"
+    yield from setup_custom_cronos(path, 26200, cfg)
+
+
+def test_mempool(cronos):
+    w3: Web3 = cronos.w3
+    filter = w3.eth.filter("pending")
+    assert filter.get_new_entries() == []
+
+    key_from = KEYS["validator"]
+    address_from = ADDRS["validator"]
+    address_to = ADDRS["community"]
+    gas_price = w3.eth.gas_price
+    cli = cronos.cosmos_cli(0)
+
+    # test contract
+    wait_for_new_blocks(cli, 1)
+    block_num_2 = w3.eth.get_block_number()
+    print(f"block number contract begin at height: {block_num_2}")
+    contract = deploy_contract(w3, CONTRACTS["Greeter"])
+    tx = contract.functions.setGreeting("world").buildTransaction()
+    signed = sign_transaction(w3, tx, key_from)
+    txhash = w3.eth.send_raw_transaction(signed.rawTransaction)
+    w3.eth.wait_for_transaction_receipt(txhash)
+    # check tx in mempool
+    new_txs = filter.get_new_entries()
```

Comment:
to test the mempool, we should try to call this before the tx is included, move this before the `wait_for_transaction_receipt`?

Reply:
this is what confused me a lot too, if I don't call `wait_for_transaction_receipt`(tx in block mined), `get_new_entries()` returns empty, even if I wait for 15 secs(before the block mined). It seems that the pending filter will not return the pending tx until the tx is mined.